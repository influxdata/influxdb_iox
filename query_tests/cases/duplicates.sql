-- Test for predicate push down explains
-- IOX_SETUP: OneMeasurementThreeChunksWithDuplicates

-- Plan with order by
explain verbose select time, state, city, min_temp, max_temp, area from h2o order by time, state, city;


-- plan without order by
explain verbose select time, state, city, min_temp, max_temp, area from h2o;

-- Union plan
EXPLAIN VERBOSE select state as name from h2o UNION ALL select city as name from h2o;
