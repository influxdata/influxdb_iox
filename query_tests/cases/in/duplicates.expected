-- Test Setup: OneMeasurementThreeChunksWithDuplicates
-- SQL: explain verbose select time, state, city, min_temp, max_temp, area from h2o order by time, state, city;
+-----------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                                                                      |
+-----------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan                            | Sort: #h2o.time ASC NULLS FIRST, #h2o.state ASC NULLS FIRST, #h2o.city ASC NULLS FIRST                                                    |
|                                         |   Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |     TableScan: h2o projection=None                                                                                                        |
| logical_plan after projection_push_down | Sort: #h2o.time ASC NULLS FIRST, #h2o.state ASC NULLS FIRST, #h2o.city ASC NULLS FIRST                                                    |
|                                         |   Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after simplify_expressions | Sort: #h2o.time ASC NULLS FIRST, #h2o.state ASC NULLS FIRST, #h2o.city ASC NULLS FIRST                                                    |
|                                         |   Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after projection_push_down | Sort: #h2o.time ASC NULLS FIRST, #h2o.state ASC NULLS FIRST, #h2o.city ASC NULLS FIRST                                                    |
|                                         |   Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after simplify_expressions | Sort: #h2o.time ASC NULLS FIRST, #h2o.state ASC NULLS FIRST, #h2o.city ASC NULLS FIRST                                                    |
|                                         |   Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| physical_plan                           | SortExec: [time@0 ASC,state@1 ASC,city@2 ASC]                                                                                             |
|                                         |   ProjectionExec: expr=[time@5 as time, state@4 as state, city@1 as city, min_temp@3 as min_temp, max_temp@2 as max_temp, area@0 as area] |
|                                         |     ExecutionPlan(PlaceHolder)                                                                                                            |
|                                         |       DeduplicateExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                               |
|                                         |         SortPreservingMergeExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                     |
|                                         |           ExecutionPlan(PlaceHolder)                                                                                                      |
|                                         |             SortExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                                |
|                                         |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                             |
|                                         |             SortExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                                |
|                                         |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                             |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                                     |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                                     |
+-----------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------+
-- SQL: explain verbose select time, state, city, min_temp, max_temp, area from h2o;
+-----------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                                                                    |
+-----------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan                            | Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |   TableScan: h2o projection=None                                                                                                        |
| logical_plan after projection_push_down | Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after simplify_expressions | Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after projection_push_down | Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| logical_plan after simplify_expressions | Projection: #h2o.time, #h2o.state, #h2o.city, #h2o.min_temp, #h2o.max_temp, #h2o.area                                                   |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                                                                    |
| physical_plan                           | ProjectionExec: expr=[time@5 as time, state@4 as state, city@1 as city, min_temp@3 as min_temp, max_temp@2 as max_temp, area@0 as area] |
|                                         |   ExecutionPlan(PlaceHolder)                                                                                                            |
|                                         |     DeduplicateExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                               |
|                                         |       SortPreservingMergeExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                     |
|                                         |         ExecutionPlan(PlaceHolder)                                                                                                      |
|                                         |           SortExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                             |
|                                         |           SortExec: [city@1 ASC NULLS LAST,state@4 ASC NULLS LAST,time@5 ASC NULLS LAST]                                                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                             |
|                                         |     IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                                     |
|                                         |     IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                                                     |
+-----------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
-- SQL: EXPLAIN VERBOSE select state as name from h2o UNION ALL select city as name from h2o;
+-----------------------------------------+---------------------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                                    |
+-----------------------------------------+---------------------------------------------------------------------------------------------------------+
| logical_plan                            | Union                                                                                                   |
|                                         |   Projection: #h2o.state AS name                                                                        |
|                                         |     TableScan: h2o projection=None                                                                      |
|                                         |   Projection: #h2o.city AS name                                                                         |
|                                         |     TableScan: h2o projection=None                                                                      |
| logical_plan after projection_push_down | Union                                                                                                   |
|                                         |   Projection: #h2o.state AS name                                                                        |
|                                         |     TableScan: h2o projection=Some([4])                                                                 |
|                                         |   Projection: #h2o.city AS name                                                                         |
|                                         |     TableScan: h2o projection=Some([1])                                                                 |
| logical_plan after simplify_expressions | Union                                                                                                   |
|                                         |   Projection: #h2o.state AS name                                                                        |
|                                         |     TableScan: h2o projection=Some([4])                                                                 |
|                                         |   Projection: #h2o.city AS name                                                                         |
|                                         |     TableScan: h2o projection=Some([1])                                                                 |
| logical_plan after projection_push_down | Union                                                                                                   |
|                                         |   Projection: #h2o.state AS name                                                                        |
|                                         |     TableScan: h2o projection=Some([4])                                                                 |
|                                         |   Projection: #h2o.city AS name                                                                         |
|                                         |     TableScan: h2o projection=Some([1])                                                                 |
| logical_plan after simplify_expressions | Union                                                                                                   |
|                                         |   Projection: #h2o.state AS name                                                                        |
|                                         |     TableScan: h2o projection=Some([4])                                                                 |
|                                         |   Projection: #h2o.city AS name                                                                         |
|                                         |     TableScan: h2o projection=Some([1])                                                                 |
| physical_plan                           | ExecutionPlan(PlaceHolder)                                                                              |
|                                         |   ProjectionExec: expr=[state@0 as name]                                                                |
|                                         |     ExecutionPlan(PlaceHolder)                                                                          |
|                                         |       ProjectionExec: expr=[state@1 as state]                                                           |
|                                         |         DeduplicateExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]           |
|                                         |           SortPreservingMergeExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST] |
|                                         |             ExecutionPlan(PlaceHolder)                                                                  |
|                                         |               SortExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                         |
|                                         |               SortExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                         |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                   |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                   |
|                                         |   ProjectionExec: expr=[city@0 as name]                                                                 |
|                                         |     ExecutionPlan(PlaceHolder)                                                                          |
|                                         |       ProjectionExec: expr=[city@0 as city]                                                             |
|                                         |         DeduplicateExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]           |
|                                         |           SortPreservingMergeExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST] |
|                                         |             ExecutionPlan(PlaceHolder)                                                                  |
|                                         |               SortExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                         |
|                                         |               SortExec: [city@0 ASC NULLS LAST,state@1 ASC NULLS LAST,time@2 ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                         |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                   |
|                                         |       IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                                   |
+-----------------------------------------+---------------------------------------------------------------------------------------------------------+
